aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -55,7 +55,9 @@ Tensor isclose(const Tensor& self, const Tensor& other, double rtol, double atol
   TORCH_CHECK(self.scalar_type() == other.scalar_type(), self.scalar_type(), " did not match ", other.scalar_type())
 
   auto actual_error = (self - other).abs();
-  auto max_error = atol + rtol * other.abs();
+  // The original formula `atol + rtol * other.abs()` works incorrectly when
+  // other has integral dtype and other = min_value and abs(min_value) is negative`.
```

Comment:
Also curious why would `abs(min_value)` even be negative?

Reply:
[If std::abs is called with an argument of type X such that std::is_unsigned<X>::value is true and X cannot be converted to int by integral promotion, the program is ill-formed.
](https://en.cppreference.com/w/cpp/numeric/math/abs)

There is no positive value corresponding to std::numeric_limits<T>::lowest() for all signed types:

`std::abs(std::numeric_limits<int64_t>::lowest()) == std::numeric_limits<int64_t>::lowest()`